### PR TITLE
DD-95: redis subscriber

### PR DESCRIPTION
**Pull request by lana-shanghai**
basic redis subscription functionality

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+use redis::{Client, Commands, Connection, RedisResult};
+use std::thread;
+mod redis_subscriber;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let client = redis::Client::open("redis://localhost:6379").unwrap();
+    let mut conn = client.get_connection().unwrap();
+
+    if let Err(error) = redis_subscriber::subscribe(String::from("external-myco/*/offers-bids/response/")) {
```
Comment (faizan2590) [inline src/main.rs:11]: I guess `psubscribe` to be used here as channel nnames have a pattern via `*`

File: src/main.rs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        println!("{:?}", error);
+        panic!("{:?}", error);
+    } else {
+        println!("connected to queue");
```
Comment (faizan2590) [inline src/main.rs:15]: message should be updated accordingly too, as you just did the subscription not the connection to queue

File: src/main.rs [hunk 3]
```diff
@@ -0,0 +1,16 @@
+extern crate redis;
+mod redis_subscriber;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let channel = String::from("external-myco/*/");
+
+    if let Err(error) = redis_subscriber::subscribe(channel.clone()) {
```
Comment (faizan2590) [inline src/main.rs:8]:
```suggestion
    if let Err(error) = redis_subscriber::psubscribe(String::from("external-myco/*/offers-bids/response/")) {
```

Can you try this?

Comment (lana-shanghai) [reply src/main.rs:8]: redis_subscriber is our internal module, and we have a function there, fn subscribe, which uses the psubscribe method :)

Comment (faizan2590) [reply src/main.rs]: Resolved after offline discussion and renaming module's method name https://github.com/gridsingularity/myco_client_rust/pull/3/commits/1330bf9eb021c476b5a78a3bcaa69fb5763ffc0b

**Review by faizan2590 — APPROVED** [review]
LGTM

**Pull request merged**